### PR TITLE
Allow user to overwrite a file if it already exists

### PR DESCRIPTION
**Pull request by volodymyrkochyn**
### Motivate of the pull request
 - [ ] To address an existing issue. If so, please provide a link to the issue: <issue id>
 - [x] Breaking new feature. If so, please describe details in the description part.

### Describe the technical details
Provide "overwrite" option when creating file with the same name in Raptor GUI

image

 ### Which part of the code base require a change
 
 - [x] Library: newproject
 - [ ] Plug-in: <Specify the plugin name>
 - [ ] Engine
 - [ ] Documentation
 - [x] Regression tests
 - [ ] Continous Integration (CI) scripts

### CODE REVIEW

File: src/NewProject/source_grid.cpp [hunk 1]
```diff
@@ -540,12 +550,54 @@ QString sourceGrid::Filter(int projectType, GridType gType) const {
...
   if (!files.isEmpty()) {
-    QMessageBox::warning(
-        this, "File(s) already added",
-        QString{"The folowing files already added to project.\n\n%1"}.arg(
-            files.join("\n")));
+    if (override) {
+      const int returnCode = QMessageBox::warning(
+          this, "File already added",
+          QString{"File %1 has already been added to the project"
+                  ". Do you want to override it?"}
```
Comment (ravikiranchollangi) [inline src/NewProject/source_grid.cpp:560]:
```suggestion
                  ". Do you want to overwrite it?"}
```

File: src/NewProject/source_grid.cpp [hunk 2]
```diff
@@ -540,12 +550,54 @@ QString sourceGrid::Filter(int projectType, GridType gType) const {
...
+              .arg(files.first()),
+          QMessageBox::Yes | QMessageBox::No | QMessageBox::Cancel);
+      return returnCode == QMessageBox::Yes;
+    } else {
+      QMessageBox::warning(this, "File(s) already added",
```
Comment (ravikiranchollangi) [inline src/NewProject/source_grid.cpp:565]:
```suggestion
      QMessageBox::warning(this, "File overwrite",
```

File: src/NewProject/source_grid.cpp [hunk 3]
```diff
@@ -540,12 +550,54 @@ QString sourceGrid::Filter(int projectType, GridType gType) const {
...
+                           QString{"The following files have already been "
```
Comment (ravikiranchollangi) [inline src/NewProject/source_grid.cpp:566]: I don't see this in screenshot.

**Review by ravikiranchollangi — CHANGES_REQUESTED** [review]
Few changes to review offline.

File: src/NewProject/source_grid.cpp (hunk 3 above)
Comment (volodymyrkochyn) [reply src/NewProject/source_grid.cpp:566]: This is old message

File: src/NewProject/source_grid.cpp (hunk 2 above)
Comment (volodymyrkochyn) [reply src/NewProject/source_grid.cpp:565]: > 

This is old message when you add few files or directory. Nothing was changed for that cases.

**Review by ravikiranchollangi — APPROVED** [review]

**Pull request merged**